### PR TITLE
Suggested edits to TsneDemo

### PR DESCRIPTION
**Pull request by mgdotdev**
1. refactored graphing protocols to satisfy the depreciation warning; pivoted towards using an ax. protocol for a more OOP graphing approach

2. In cell #5 for coordinates_and_bg, used transpose() on an arrayed list of lists, instead of greating a zero array and populating the array piecewise.

3. Instead of using for loops over print functions, changed the display protocol to use list comprehensions in dictionaries, then pushed the dictionaries into a set of pandas dataframes for better display. Also allows for hiding the URL string in an HTML hyperlink.

4. Added appropriate comments to subsection the jupyter cells for user convenience, and fixed inconsistent naming conventions.

### CODE REVIEW

**Comment by mgdotdev** [comment]
Obviously the .travis.yml check is going to fail since I'm not providing my personal API key; how would y'all like the CitrinationClient(api_key=?) to be formatted to demonstrate the CI build pass? Notebook works fine locally, I'm just not pushing a personal api_key to GitHub.

**Review by ahl389 — DISMISSED** [review]
Sounds good to me. I don't have any familiarity with this project, but I'm happy to see these changes merged if others have deemed them necessary!

File: citrination_api_examples/TsneDemo.ipynb [hunk 1]
```diff
@@ -72,12 +74,15 @@
     }
    ],
    "source": [
-    "# Initialize the CitrinationClient\n",
-    "site = \"https://citrination.com\" # public\n",
-    "client = CitrinationClient(api_key=os.environ[\"CITRINATION_API_KEY\"], site=site)\n",
+    "# Initialize the CitrinationClient \n",
+    "client = CitrinationClient(\n",
+    "    api_key=\"INSERT_PERSONAL_API_KEY\",\n",
```
Comment (maxhutch) [inline citrination_api_examples/TsneDemo.ipynb]:
```suggestion
    "    api_key=os.environ[\"CITRINATION_API_KEY\"],\n",
```

**Review by maxhutch — CHANGES_REQUESTED** [review]
> Obviously the .travis.yml check is going to fail since I'm not providing my personal API key; how would y'all like the CitrinationClient(api_key=?) to be formatted to demonstrate the CI build pass? Notebook works fine locally, I'm just not pushing a personal api_key to GitHub.

The testing harness inserts a valid API key into the environment, so if you change that block back to th way it was, then the tests will pass.  It is a dangerous practice to ever put any API key (or password) in code, which is why we strongly recommend putting it into your environment and loading it using `os.environ`.

**Comment by maxhutch** [comment]
> 1. refactored graphing protocols to satisfy the depreciation warning; pivoted towards using an ax. protocol for a more OOP graphing approach
> 2. In cell #5 for coordinates_and_bg, used transpose() on an arrayed list of lists, instead of greating a zero array and populating the array piecewise.
> 3. Instead of using for loops over print functions, changed the display protocol to use list comprehensions in dictionaries, then pushed the dictionaries into a set of pandas dataframes for better display. Also allows for hiding the URL string in an HTML hyperlink.
> 4. Added appropriate comments to subsection the jupyter cells for user convenience, and fixed inconsistent naming conventions.

These changes are slick!  Assuming the tests pass after you update the client definition block, it should be good to go.

**Comment by maxhutch** [comment]
https://github.com/CitrineInformatics/learn-citrination/pull/37 will fix the python 2.x failure.  To pickup the API key, we need to move this branch into `CitrineInformatics/learn-citrination` (rather than your fork).  I'm going to create a new target branch for you, do the code review on the way into it, and then open a follow-up PR to get it into `master`.

========
(edited)
See https://docs.travis-ci.com/user/environment-variables/ :
```
Encrypted environment variables are not available to pull requests from forks due to the security risk of exposing such information to unknown code.

If you define a variable with the same name in .travis.yml and in the Repository Settings, the one in .travis.yml takes precedence. If you define a variable in .travis.yml as both encrypted and unencrypted, the one defined later in the file takes precedence.
```

**Comment by mgdotdev** [comment]
Sounds good, just let me know which branch you'd like to pull `1mikegrn/learn-citrination` towards and I'll set up the PR to push into it.

**Review by maxhutch — APPROVED** [review]

**Comment by maxhutch** [comment]
We expect the tests to pass once the branch is staged.

**Pull request merged**